tsdb/isolation.go

```diff
@@ -33,6 +33,10 @@ type isolationState struct {
 
 // Close closes the state.
 func (i *isolationState) Close() {
+	if i.isolation == nil || i.isolation.disabled {
```

Comment:
Suspect this may be wrong - we add to the `readsOpen` list in `isolation.State()`, so we should remove from the list here.
https://github.com/prometheus/prometheus/pull/9270/files#diff-222a4dbb32642e9d05ef238f09f69f2d1f0e86c211aa0beaf071dc20a1d96c90R136-R137

Bonus points (if I'm right): figure out which test should have told you this.

Reply:
You are right, we need to close this

Reply:
`TestWaitForPendingReadersInTimeRange` catches this, since it depends on the read tracking. Fails with this, works when removed.